DNN Platform/Dnn.AuthServices.Jwt/Components/Common/Controllers/JwtController.cs

```diff
@@ -224,17 +223,29 @@ public LoginResultData RenewToken(HttpRequestMessage request, string renewalToke
                 return EmptyWithError("not-more-renewal");
             }
 
+            if (ptoken.RenewalHash != GetHashedStr(renewalToken))
+            {
+                if (Logger.IsTraceEnabled) Logger.Trace("Invalid renewal token");
+                return EmptyWithError("bad-renewal-token");
+            }
+
+            if (ptoken.TokenHash != GetHashedStr(rawToken))
+            {
+                if (Logger.IsTraceEnabled) Logger.Trace("Invalid access token");
+                return EmptyWithError("bad-access-token");
+            }
+
             var userInfo = TryGetUser(jwt, false);
             if (userInfo == null)
             {
-                if (Logger.IsTraceEnabled) Logger.Trace("Token not found in DB");
-                return EmptyWithError("not-found");
+                if (Logger.IsTraceEnabled) Logger.Trace("User not found in DB");
+                return EmptyWithError("user-not-found");
             }
 
-            if ((ptoken.TokenHash != GetHashedStr(rawToken)) || (ptoken.UserId != userInfo.UserID))
+            if ((ptoken.UserId != userInfo.UserID))
             {
-                if (Logger.IsTraceEnabled) Logger.Trace("Mismatch in received token");
-                return EmptyWithError("bad-token");
+                if (Logger.IsTraceEnabled) Logger.Trace("Mismatch token and user");
+                return EmptyWithError("bad-token-user");
```

Comment:
Same question here, I'm inclined to say it should just be "bad-token"

Reply:
var secret = ObtainSecret(sessionId, portalSettings.GUID, userInfo.Membership.LastPasswordChangeDate);

secret is compliantly unique for each user in each portal so I think there is no reason for worry about the disclose the details.